mypy/semanal.py

```diff
@@ -3638,9 +3653,26 @@ def lookup_qualified(self, name: str, ctx: Context,
                     if n and n.module_hidden:
                         self.name_not_defined(name, ctx)
             if n and not n.module_hidden:
+                n = self.rebind_symbol_table_node(n)
                 return n
             return None
 
+    def rebind_symbol_table_node(self, n: SymbolTableNode) -> Optional[SymbolTableNode]:
+        """If node refers to old version of module, return reference to new version.
+
+        If the reference is removed in the new version, return None.
+        """
+        # TODO: Handle type aliases, type variables and other sorts of references
+        if isinstance(n.node, (FuncDef, OverloadedFuncDef, TypeInfo, Var)):
+            # TODO: Why is it possible for fullname() to be None, even though it's not
+            #   annotated as Optional[str]?
```

Comment:
I think this is because of pre-PEP 526 limitations for ``_fullname`` (note underscore) and the fact that ``_fullname`` is not set in ``__init__`` but later (first vs second pass IIRC). We already had few crashes because of this. I fixed some of them but maybe some nodes still don't have full names generated in second pass.